flytekit/core/promise.py

```diff
@@ -895,12 +906,21 @@ def binding_from_python_std(
     t_value_type: type,
 ) -> Tuple[_literals_models.Binding, List[Node]]:
     nodes: List[Node] = []
-    binding_data = binding_data_from_python_std(
-        ctx,
-        expected_literal_type,
-        t_value,
-        t_value_type,
-        nodes,
+    try:
+        asyncio.get_running_loop()
+        raise AssertionError("binding_from_python_std cannot be run from within an async call stack")
+    except RuntimeError as e:
+        if "no running event loop" not in str(e):
+            logger.error(f"Unknown RuntimeError {str(e)}")
+            raise
+    binding_data = asyncio.run(
```

Comment:
What is the issue with using `sync_run` here?

Reply:
updated